src/eventHandlers/FunctionsMetadataHandler.ts

```diff
@@ -27,10 +28,12 @@ export class FunctionsMetadataHandler extends EventHandler<'functionsMetadataReq
             logCategory: LogCategory.System,
         });
 
-        const functions = Object.values(worker.app.functions);
-        if (functions.length > 0) {
-            response.useDefaultMetadataIndexing = false;
-            response.functionMetadataResults = functions.map((f) => f.metadata);
+        if (worker.app.isUsingWorkerIndexing) {
+            if (isDefined(worker.app.blockingAppStartError)) {

```

Comment:
Why don't we throw `worker.app.blockingAppStartError` anyway, regardless of whether `worker.app.isUsingWorkerIndexing`?